frontend/src/components/OrderPage.js

```diff
@@ -113,6 +120,27 @@ class OrderPage extends Component {
     this.getOrderDetails(this.state.orderId);
   }
 
+  handleWebLN = (data) => {
+    if (data.status !== this.state.status) {
+      getWebLN()
+        .then((webln) => {
+          // If WebLN implements locked payments compatibility, this logic won't be necesary anymore
+          if (this.state.is_maker & (data.status == 0 || data.status == 1)) {
+            webln.sendPayment(this.state.bond_invoice);
+            this.setState({ waitingWebNL: true, openWebNLDialog: true});
+          } else if (this.state.is_taker & data.status == 3) {
+            webln.sendPayment(this.state.bond_invoice);
+            this.setState({ waitingWebNL: true, openWebNLDialog: true});
+          } else if (this.state.is_seller & (data.status == 6 || data.status == 7 )) {
+            webln.sendPayment(this.state.escrow_invoice);
+            this.setState({ waitingWebNL: true, openWebNLDialog: true});
+          } else {
+            this.setState({ waitingWebNL: false });
+          }
+        });
+  }
+}
```

Comment:
This is the only way I found to manage `status` changes without using Hooks, happy to hear any other suggestions I might have missed out.

Reply:
Please also confirm this is the right combination of `user rol + status`